### ISSUE TITLE
Move `run_if` to `IntoSystem`

### ISSUE THREAD

**Issue by DasLixou** [bevyengine/bevy#14576]
## What problem does this solve or what need does it fill?

I am using `bevy_eventlistener` to listen for events and want to have a run condition on the runned system.
However, `run_if` is only implemented for `IntoSystemConfigs`, not `IntoSystem`, where only `pipe` and `map` exist, which both don't really help me.

## What solution would you like?

Move / implement `run_if` for `IntoSystem`

## What alternative(s) have you considered?

Hand-wiring the two systems together.

**Comment by alice-i-cecile** [comment]
Run conditions still need to work on system sets. If anyone tries their hand at this change, that's the critical requirement that I don't think will be solved by simply moving the method.

**Comment by DasLixou** [comment]
Rust compiler would nag when both `IntoSystemConfigs` and `IntoSystem` had a method called `run_if`, right? :c

**Comment by alice-i-cecile** [comment]
Yeah, I think it would likely end up ambiguous, which is no fun at all and would seriously degrade UX. By the way, `bevy_eventlistener` is largely doomed with #13991 being merged, but we still need run conditions on observers anyways.

**Comment by DasLixou** [comment]
I guess `bevy_mod_picking` will also not receive an update using observers because of it's upstreaming? So waiting for 0.15 release it is..?

**Comment by alice-i-cecile** [comment]
Most likely yeah. Fingers crossed we can get everything in place for 0.15 though!

**Comment by DasLixou** [comment]
Good news! I could work around it pretty easy by pipeing the bool into my system as an `In` and returning early

**Comment by jnhyatt** [comment]
> we still need run conditions on observers anyways

Is this the case if we get entity disabling?

**Comment by ItsDoot** [comment]
Glad you found a solution! I just don't see how we could move it without compromising our current feature set.

**Comment by ItsDoot** [comment]
On second thought, I think we could accomplish this with some new `filter` function, akin to how `Iterator::filter` works:

```rust
pub trait IntoSystem<In: SystemInput, Out, Marker>: Sized {
    type System: System<In = In, Out = Out>;

    fn filter<T, F>(self, f: F) -> IntoFilteredSystem<Self, F>
    where
        F: Send + Sync + 'static + Fn(&Out) -> bool;

    // ...
}
```

**Comment by targrub** [comment]
@ItsDoot See merged PR https://github.com/bevyengine/bevy/pull/22602
